src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/custom.py

```diff
@@ -143,44 +143,50 @@ def enable_local_git(resource_group, name, slot=None):
 
     return {'url' : _get_git_url(client, resource_group, name, slot)}
 
-def create_app_service_plan(resource_group, name, tier=None, number_of_workers=None,
+def create_app_service_plan(resource_group, name, sku, number_of_workers=None,
                             location=None):
     client = web_client_factory()
+    sku = _normalize_sku(sku)
     if location is None:
         location = _get_location_from_resource_group(resource_group)
 
-    sku_name = _get_sku_name(tier)
     #the api is odd on parameter naming, have to live with it for now
-    sku = SkuDescription(name=tier, tier=sku_name, capacity=number_of_workers)
-    plan_def = ServerFarmWithRichSku(location, server_farm_with_rich_sku_name=name, sku=sku)
-    #TODO: handle bad error on creating too many F1 plans:
-    #  Operation failed with status: 'Conflict'. Details: 409 Client Error: Conflict for url:
+    sku_def = SkuDescription(tier=_get_sku_name(sku), name=sku, capacity=number_of_workers)
+    plan_def = ServerFarmWithRichSku(location, server_farm_with_rich_sku_name=name, sku=sku_def)
     return client.server_farms.create_or_update_server_farm(resource_group, name, plan_def)
 
-def update_app_service_plan(resource_group, name, tier=None, number_of_workers=None,
+def update_app_service_plan(resource_group, name, sku=None, number_of_workers=None,
```

Comment:
We should also wrap this custom command in generic update to maintain consistency with our other update commands (and to allow, for example, tagging).